### PR TITLE
Add dataset reader for DROP

### PR DESCRIPTION
**Pull request by matt-gardner**

### CODE REVIEW

**Comment by matt-gardner** [comment]
@pdasigi, don't feel like you need to give this a super careful look - @yizhongw wrote it, and I've already looked over it some, so it's had two eyes already.  If you _want_ to give it a careful look, that's fine too.

File: allennlp/data/dataset_readers/reading_comprehension/util.py [hunk 1]
```diff
@@ -369,3 +369,41 @@ def handle_cannot(reference_answers: List[str]):
     else:
         reference_answers = [x for x in reference_answers if x != 'CANNOTANSWER']
     return reference_answers
+
+
+def split_token_by_hyphen(token, hyphen):
```
Comment (pdasigi) [inline allennlp/data/dataset_readers/reading_comprehension/util.py]: Looks like this function is supposed to work for any delimiter. `split_token_by_delimiter` seems more appropriate.

Comment (pdasigi) [reply allennlp/data/dataset_readers/reading_comprehension/util.py]: Also add type hints for arguments and return values.

File: allennlp/data/dataset_readers/reading_comprehension/util.py [hunk 2]
```diff
@@ -369,3 +369,41 @@ def handle_cannot(reference_answers: List[str]):
...
+        char_offset += len(hyphen)
+    if split_tokens:
+        split_tokens.pop(-1)
+        char_offset -= len(hyphen)
+        return split_tokens
+    else:
+        return [token]
+
+
+def split_tokens_by_hyphen(tokens):
```
Comment (pdasigi) [inline allennlp/data/dataset_readers/reading_comprehension/util.py]: Missing type hints.

**Comment by matt-gardner** [comment]
Thanks for the comments!  Can you look over the docstring for the dataset reader again and see if it makes more sense?

**Comment by pdasigi** [comment]
@matt-gardner Yes, the docstring looks good now.

**Pull request merged**